src/student/student.service.ts

```diff
@@ -1,29 +1,91 @@
-import { Injectable } from "@nestjs/common";
+import { Injectable, HttpException } from "@nestjs/common";
 import { StudentInterface } from "./interfaces/student.interface";
 import { HttpService } from "@nestjs/axios";
 import { AxiosResponse } from "axios";
-import { first, map, Observable } from "rxjs";
+import { first, map, catchError, Observable } from "rxjs";
 import { response } from "express";
+import { StudentDto } from "./dto/student.dto";
+import { SuccessResponse } from "src/success-response";
+import { ErrorResponse } from "../error-response";
+import { SaveStudentDto } from "./dto/save-student.dto";
 
+var objectMapper = require("object-mapper");
 @Injectable()
 export class StudentService {
   private student: StudentInterface;
 
   constructor(private httpService: HttpService) {}
 
-  getOne(id: any): Observable<StudentInterface> {
-    return this.httpService
-      .get(
-        "https://dev-shiksha.uniteframework.io/registry/api/v1/Student/c008ac28-cbc1-4abb-a215-4917e360b1a1"
-      )
-      .pipe(
-        map((axiosResponse: AxiosResponse) => {
-          this.student = {
-            studentId: id,
-            firstName: axiosResponse.data.studentFirstName,
-          };
-          return this.student;
-        })
-      );
+  url = "https://dev-shiksha.uniteframework.io/registry/api/v1/Student";
+  public async getStudent(id: any, request: any) {
+    return this.httpService.get(`${this.url}/${id}`, request).pipe(
+      map((axiosResponse: AxiosResponse) => {
+        const data = axiosResponse.data;
+
+        let student = {
+          osid: data.osid,
```

Comment:
osid is not needed for Shiksha. Thr osid from Sunbird RC should be mapped to shiksha's studentId